server/games/game.py

```diff
@@ -358,26 +358,28 @@ def add_game_connection(self, game_connection):
             elif self.state == GameState.LIVE:
                 self._logger.info("Game finished normally")
 
-                if self.desyncs > 20:
-                    await self.mark_invalid(ValidityState.TOO_MANY_DESYNCS)
-                    return
-
-                if time.time() - self.launched_at > 4*60 and self.is_mutually_agreed_draw:
-                    self._logger.info("Game is a mutual draw")
-                    await self.mark_invalid(ValidityState.MUTUAL_DRAW)
-                    return
-
-                if len(self.players) < 2:
-                    await self.mark_invalid(ValidityState.SINGLE_PLAYER)
-                    return
-
-                if len(self._results) == 0:
-                    await self.mark_invalid(ValidityState.UNKNOWN_RESULT)
-                    return
-
-                await self.persist_results()
-                await self.rate_game()
-                await self._process_pending_army_stats()
+                if self.validity is ValidityState.VALID:
```

Comment:
You should invert your logic which makes it easier to read.
```suggestion
                if self.validity is not ValidityState.VALID:
                    return
```